xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm

```diff
@@ -141,7 +141,11 @@ $xwiki.ssfx.use('uicomponents/viewers/comments.css', true)
   #if ("$!commentSelection" != "")
     #set($isAnnotation = true)
   #end
-  #set ($isUserComment = $services.model.resolveDocument($comment.author, 'current') == $xcontext.userReference)
+  #if ($comment.author == '')
+    #set ($isUserComment = false)
+  #else
+    #set ($isUserComment = $services.model.resolveDocument($comment.author, 'user') == $xcontext.userReference)
+  #end
```

Comment:
Could be written on a single line I think

    #set ($isUserComment = $comment.author != '' && $services.model.resolveDocument($comment.author, 'user') == $xcontext.userReference)

I would add a comment explaining that empty author means guest user and why we don't allow guest users to edit or delete annotations.

Reply:
Indeed.